packages/marketplace/contracts/transfer-manager/TransferManager.sol

```diff
@@ -177,46 +161,36 @@ abstract contract TransferManager is ERC165Upgradeable, ITransferManager {
     /// @param amount total amount of asset that is going to be transferred
     /// @param from owner of the Asset to transfer
     /// @return How much left after transferring royalties
-    function transferRoyalties(
+    function _transferRoyalties(
         LibAsset.AssetType memory paymentAssetType,
         LibAsset.AssetType memory nftAssetType,
         LibPart.Part[] memory payouts,
         uint256 rest,
         uint256 amount,
         address from
     ) internal returns (uint256) {
-        LibPart.Part[] memory royalties = getRoyaltiesByAssetType(nftAssetType);
+        LibPart.Part[] memory royalties = _getRoyaltiesByAssetType(nftAssetType);
 
-        if (
-            nftAssetType.assetClass == LibAsset.AssetClassType.ERC1155_ASSET_CLASS ||
-            nftAssetType.assetClass == LibAsset.AssetClassType.ERC721_ASSET_CLASS
-        ) {
-            (address token, uint256 tokenId) = abi.decode(nftAssetType.data, (address, uint));
-            try IERC165Upgradeable(token).supportsInterface(INTERFACE_ID_IROYALTYUGC) returns (bool resultInterface) {
-                if (resultInterface) {
-                    address creator = IRoyaltyUGC(token).getCreatorAddress(tokenId);
-                    if (payouts.length == 1 && payouts[0].account == creator) {
-                        require(royalties[0].value <= 5000, "Royalties are too high (>50%)");
-                        return rest;
-                    }
-                }
-                // solhint-disable-next-line no-empty-blocks
-            } catch {}
-        }
-        if (royalties.length == 1 && payouts.length == 1 && royalties[0].account == payouts[0].account) {
-            require(royalties[0].value <= 5000, "Royalties are too high (>50%)");
-            return rest;
+        if (payouts.length == 1) {
+            address creator = _getCreator(nftAssetType);
+            if (creator != address(0) && payouts[0].account == creator) {
```

Comment:
aren't we missing this
```royalties.length == 1 && ```

Reply:
it is in line180, I just surrounded everything inside the  `if (payouts.length == 1) `